From a77110745cf8e0367c202b82e53570aa28662ddc Mon Sep 17 00:00:00 2001
From: bodapati <shalini.salomi.bodapati@intel.com>
Date: Sat, 19 Sep 2020 19:40:25 +0530
Subject: [PATCH] Build Support to enable avx

This patch removes -march=skylake by default in
build system. Individual modules need to enable it based on
perf characterization.

Change-Id: I663b0fdab5663d4111fac37df52c164ff1630db8
Tracked-On: OAM-92975
Signed-off-by: bodapati <shalini.salomi.bodapati@intel.com>
---
 android/arch.go              | 2 ++
 cc/config/x86_64_device.go   | 3 ++-
 cc/config/x86_device.go      | 4 +++-
 rust/config/x86_64_device.go | 1 +
 4 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/android/arch.go b/android/arch.go
index d14221f..7ae1b29 100644
--- a/android/arch.go
+++ b/android/arch.go
@@ -1677,6 +1677,7 @@ func getMegaDeviceConfig() []archConfig {
 		{"x86", "ivybridge", "", []string{"x86"}},
 		{"x86", "sandybridge", "", []string{"x86"}},
 		{"x86", "silvermont", "", []string{"x86"}},
+		{"x86", "kabylake", "", []string{"x86"}},
 		{"x86", "stoneyridge", "", []string{"x86"}},
 		{"x86", "x86_64", "", []string{"x86"}},
 		{"x86_64", "", "", []string{"x86_64"}},
@@ -1684,6 +1685,7 @@ func getMegaDeviceConfig() []archConfig {
 		{"x86_64", "ivybridge", "", []string{"x86_64"}},
 		{"x86_64", "sandybridge", "", []string{"x86_64"}},
 		{"x86_64", "silvermont", "", []string{"x86_64"}},
+		{"x86_64", "kabylake", "", []string{"x86_64"}},
 		{"x86_64", "stoneyridge", "", []string{"x86_64"}},
 	}
 }
diff --git a/cc/config/x86_64_device.go b/cc/config/x86_64_device.go
index bcfae5d..b08ee9f 100644
--- a/cc/config/x86_64_device.go
+++ b/cc/config/x86_64_device.go
@@ -55,8 +55,9 @@ var (
 			"-march=slm",
 		},
 		"skylake": []string{
-			"-march=skylake",
 		},
+                "kabylake": []string{
+                },
 		"stoneyridge": []string{
 			"-march=bdver4",
 		},
diff --git a/cc/config/x86_device.go b/cc/config/x86_device.go
index 64392dc..8fcadef 100644
--- a/cc/config/x86_device.go
+++ b/cc/config/x86_device.go
@@ -71,8 +71,10 @@ var (
 			"-march=slm",
 			"-mfpmath=sse",
 		},
+                "kabylake": []string{
+                         "-mfpmath=sse",
+                },
 		"skylake": []string{
-			"-march=skylake",
 			"-mfpmath=sse",
 		},
 		"stoneyridge": []string{
diff --git a/rust/config/x86_64_device.go b/rust/config/x86_64_device.go
index 9a6c00b..7cb6d7d 100644
--- a/rust/config/x86_64_device.go
+++ b/rust/config/x86_64_device.go
@@ -33,6 +33,7 @@ var (
 		"sandybridge": []string{"-C target-cpu=sandybridge"},
 		"silvermont":  []string{"-C target-cpu=silvermont"},
 		"skylake":     []string{"-C target-cpu=skylake"},
+		"kabylake":    []string{"-C target-cpu=kabylake"},
 		//TODO: Add target-cpu=stoneyridge when rustc supports it.
 		"stoneyridge": []string{""},
 	}
-- 
2.7.4

